server/server_test.go

```diff
@@ -274,6 +276,121 @@ func TestSubmitCertificateVerification(t *testing.T) {
 	require.NoError(t, eg.Wait())
 }
 
+func createTestKeyFile(t *testing.T, dir, name string, data []byte) string {
+	path := filepath.Join(dir, name)
+	if err := os.WriteFile(path, data, 0o644); err != nil {
+		t.Fatalf("failed to write key file: %v", err)
+	}
+	return path
+}
+
+// Test submitting a challenge followed by proof generation and getting the proof via GRPC.
+func TestLoadTrustedKeysAndSubmit(t *testing.T) {
```

Comment:
It would be great to also add a test to verify if previously configured trusted keys are forgotten after their files were removed and keys were reloaded.

Reply:
Done